src/Symfony/Component/Routing/Matcher/UrlMatcher.php

```diff
@@ -223,7 +233,13 @@ protected function getAttributes(Route $route, string $name, array $attributes):
     protected function handleRouteRequirements(string $pathinfo, string $name, Route $route): array
     {
         // expression condition
-        if ($route->getCondition() && !$this->getExpressionLanguage()->evaluate($route->getCondition(), ['context' => $this->context, 'request' => $this->request ?: $this->createRequest($pathinfo)])) {
+        if (
+            $route->getCondition() &&
+            !$this->getExpressionLanguage()->evaluate($route->getCondition(), array_merge($this->expressionConditionVariables, [
```

Comment:
no need to call array_merge, a `+` would do it

Reply:
array merging isn't necessary anymore because of changes in eafc9c5749ac2063e5d9c3689a19202b12cbc6d1